### DISCUSSION TITLE
Inquiry About DDS and SOME/IP Gateway Integration with Iceoryx

### DISCUSSION THREAD

**Discussion by Abhishek2581** [eclipse-iceoryx/iceoryx#2375] [Q&A]
Hello Team,

While exploring the Iceoryx code, I noticed that it provides options to implement DDS and/or SOME/IP gateways for communication. I also came across the iceoryx-gateway-dds project, but I found that it is not buildable, making it less useful for understanding the gateway functionality.

Is there any documentation available that explains how to implement these gateways and integrate them with Iceoryx? Alternatively, a proof-of-concept or example demonstrating this integration would be highly helpful.

Looking forward to your guidance.

Best regards,
Abhishek GOYAL

**Comment by elfenpiff** [comment]
Hi @Abhishek2581 

the iceoryx-gateway-dds project went a bit out-of-date, but it should still be working with older iceoryx versions. Maybe @elBoberido or @orecham have more insights here.
Btw. we are offering commercial support for iceoryx, contact us via info@ekxide.io if you are interested. For instance, we could update the iceoryx-gateway-dds project to be compatible with the iceoryx version you are using.

We have at least a gateway library, as mentioned here: https://github.com/eclipse-iceoryx/iceoryx/blob/main/doc/website/concepts/architecture.md

Currently, we are focusing on [iceoryx2](https://github.com/eclipse-iceoryx/iceoryx2) and finishing the last missing pieces to achieve feature parity with iceoryx, which will include gateways.

Would you be able to share your use case with us?

**Comment by Abhishek2581** [comment]
Hello Christian,

Thank you for the prompt response and the detailed insights!

We understand that the iceoryx-gateway-dds project is currently outdated and may require older versions of Iceoryx to function. While commercial support is an interesting option, my immediate goal is to evaluate the gateway functionality for potential use in our projects, particularly in the context of Software-Defined Vehicles (SDV).

To achieve this, We aim to:
1. Understand how gateways can be implemented using Iceoryx.
2. Evaluate the performance and integration of DDS or SOME/IP gateways with Iceoryx in an SDV context.
3. Explore whether this architecture aligns with our requirements for future projects.

We have reviewed the gateway library documentation referenced in the architecture guide, and We believe that a working proof-of-concept or example implementation of a gateway, even with an older version of Iceoryx, would be highly valuable for our exploration.

Could you please provide guidance on:
1. The specific Iceoryx version compatible with iceoryx-gateway-dds?
2. Any resources or references that would help me understand the current implementation of the gateway library and how it could be extended or adapted?

Additionally, I would be happy to share more details about our use case if it helps provide better context for your guidance.

Looking forward to your support.

Best Regards,
Abhishek GOYAL

**Reply by elfenpiff** [reply]
Pinging @orecham . He implemented the iceoryx-gateway-dds and may be able to provide you the necessary details.

**Reply by Abhishek2581** [reply]
Thanks Christian

**Reply by orecham** [reply]
@Abhishek2581 The best resource is the code in `iceoryx-gateway-dds`. It's not a large code base so it should be quick to grasp.

As for the version, I asked you a follow-up to confirm the version in your issue, but I assume there is a branch of `iceoryx` which can build because someone added a fast dds gateway just last month.

**Reply by elBoberido** [reply]
@Abhishek2581 would a zenoh gateway also be an option for you? Our main focus shifts to iceoryx2, which should surpass classic iceoryx in terms on features the next 2-3 months. The initial iceoryx2 gateway will probably be based on zenoh.

Can you share more details about your use case? If you cannot do it in public, you can check my github profile on how to contact me.
